subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h

```diff
@@ -1279,6 +1276,15 @@ static inline void hal_radio_sw_switch_disable(void)
 	HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI_REGISTER_TASK(1) = 0;
 }
 
+static inline void hal_radio_sw_switch_cleanup(void)
+{
+	hal_radio_sw_switch_disable();
```

Comment:
@cvinayak if i understand correctly, _this_ is not part of the current implementation. So, now, in radio_disable(), nrf53 will not only disable the 2 DPPI channels, but also clear any task subscribe registers. 

Seems no harm, but wanted to double-check with you before merging

Reply:
> now, in radio_disable(), nrf53 will not only disable the 2 DPPI channels, but also clear any task subscribe registers.

This is intentional. Currently, the `hal_radio_sw_switch_disable` function is not consistent between nRF51/nRF52 and nRF53. For nRF53, the `nrf_dppi_channels_disable(NRF_DPPIC,...)` call ideally should have to be in `hal_radio_sw_switch_disable`, but you have already explained so in the comments in that function. I did not want to change, hence added disable of DPPI and reset of subscribe registers in the cleanup (this is not necessary but consistent between `_complete_and_disable` versus `radio_disable` under cases of abort. 

This was to some extent discussed with @kruithofa, who has create a issue to follow up (https://github.com/zephyrproject-rtos/zephyr/issues/26118)

Reply:
Thanks for explaining. Yes, seems not required. Consistency is fine, but will cost a little bit in code footprint and runtime overhead :)